UPGRADE-GUIDE.md

```diff
@@ -0,0 +1,460 @@
+# Upgrade Guide
+
+This document provides a full guide for upgrading a Cosmos SDK chain from `v0.50.x` to `v0.53.x`.
+
+After completing this guide, applications will have:
+
+- The `x/protocolpool` module
+- The `x/epochs` module
+- Unordered Transaction support
+
+## Table of Contents
+
+- [App Wiring Changes](#app-wiring-changes)
+- [Adding ProtocolPool Module](#adding-protocolpool-module)
+    - [Manual Wiring](#protocolpool-manual-wiring)
+    - [DI Wiring](#protocolpool-di-wiring)
+- [Adding Epochs Module](#adding-epochs-module)
+    - [Manual Wiring](#epochs-manual-wiring)
+    - [DI Wiring](#epochs-di-wiring)
+- [Enable Unordered Transactions](#enable-unordered-transactions)
+- [Upgrade Handler](#upgrade-handler)
+
+## App Wiring Changes
+
+The `x/auth` module now contains a `PreBlocker` that _must_ be set in the module manager's `SetOrderPreBlockers` method.
+
+```go
+app.ModuleManager.SetOrderPreBlockers(
+    upgradetypes.ModuleName,
+    authtypes.ModuleName, // NEW
+)
+```
+
+## Adding ProtocolPool Module
+
+### Manual Wiring
+
+Import the following:
+
+```go
+import (
+	// ...
+    "github.com/cosmos/cosmos-sdk/x/protocolpool"
+    protocolpoolkeeper "github.com/cosmos/cosmos-sdk/x/protocolpool/keeper"
+    protocolpooltypes "github.com/cosmos/cosmos-sdk/x/protocolpool/types"
+)
+```
+
+Set the module account permissions.
+
+```go
+	maccPerms = map[string][]string{
+		// ...
+		protocolpooltypes.ModuleName:                nil,
+		protocolpooltypes.ProtocolPoolEscrowAccount: nil,
+	}
+```
+
+Add the protocol pool keeper to your application struct.
+
+```go
+	ProtocolPoolKeeper protocolpoolkeeper.Keeper
+```
+
+Add the store key:
+
+```go
+keys := storetypes.NewKVStoreKeys(
+		// ...
+		protocolpooltypes.StoreKey,
+	)
+```
+
+Instantiate the keeper:
+
+```go
+app.ProtocolPoolKeeper = protocolpoolkeeper.NewKeeper(
```

Comment:
we should note that you should instantiate this before distribution so you can pass it to the distribution keeper

Reply:
added